libs/fn/src/lib/input/input.component.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { InputComponent } from './input.component';
+import { InputState } from './input-base';
+
+describe('InputComponent', () => {
+    let component: InputComponent;
+    let fixture: ComponentFixture<InputComponent>;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            declarations: [InputComponent]
+        }).compileComponents();
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(InputComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should appropriate states', () => {
+        const states: InputState[] = ['positive', 'negative', 'critical', 'info'];
+
+        states.forEach((state) => {
+            component.state = state;
+            fixture.detectChanges();
+
+            expect(fixture.nativeElement.classList).toContain(`fn-input--${state}`);
+        });
+    });
+
+    it('should writeValue', () => {
+        component.writeValue('test');
+        spyOn(component, 'onTouched');
+        spyOn(component, 'onChange');
+        expect(component.inputText).toBe('test');
+        expect(component.onTouched).toHaveBeenCalled();
+        expect(component.onChange).toHaveBeenCalled();
+    });
+
+    it('should setDisabledState', () => {
+        component.setDisabledState(true);
+        expect(component.disabledByForm).toBeTruthy();
+    });
+});
```

Comment:
can you please add `ngx-cva-test-suite` to this component? just to ensure it works correctly with angular forms. you may refer `libs/platform/src/lib/form/input/input.component.spec.ts`